server/backend/api/app/routers/router_user_role_assignment.py

```diff
@@ -18,11 +21,16 @@
 
 router = APIRouter()
 
-@router.post("",
+
+@router.post(
+    "",
     response_model=schemas.FamUserRoleAssignmentGet,
     # Guarding endpoint with Depends().
     dependencies=[
-        Depends(authorize_by_application_role),
+        Depends(authorize_by_application_role),  # requester needs to be app admin or delegated admin
+        Depends(authorize_by_privilege),  # if requester is delegated admin, needs to have privilge to grant access with the request role
+        Depends(authorize_by_user_type),  # check business bceid user cannot grant idir user access
+        Depends(enforce_bceid_by_same_org_guard),   # check business bceid user can only grant access for the user from same organization
```

Comment:
Not sure the `dependencies` for the order of each `Depends`, it could matter, I can't seem to find the doc mentioned this.
I hope it is the **natural ordering** of the Array (left to right). So, "enforce_self_grant_guard" might be the first to guard and comes the `authorize_by_application_role`, `authorize_by_user_type`, `authorize_by_privilege` then `enforce_bceid_by_same_org_guard` I guess.
I was wondering that, but if manual testing work, maybe in this case it does not have issue.

Reply:
I think I tried before, the dependencies is in order, like it will return permission error first, and then the user type error